doc/decisions/deferred_plugin_calls.md

```diff
@@ -0,0 +1,61 @@
+# Deferred Plugin Calls
+
+## Issue
+
+Since some plugins may lazy-load encapsulated plugins at a later time a
+mechanism for deferring function calls to this encapsulated plugins is needed.
+
+For example when setting I/O bindings with `elektraIoSetBinding()` the exported
+function `setIoBinding` is called for all globally mounted plugins.
+The "list" does not export this functions and cannot implement every exported
+function for its encapsulated plugins.
```

Comment:
You should also give other examples, e.g. dini and multifile, and explain if your decision also applies to them.

At the moment the decision is too much centered to your use case.